src/dotnet-install.ps1

```diff
@@ -1100,17 +1100,24 @@ function Resolve-AssetName-And-RelativePath([string] $Runtime) {
 }
 
 function Prepare-Install-Directory {
+    $diskSpaceWarning = "Failed to check the disk space. Installation will continue, but it may fail if you do not have enough disk space.";
+    # this check is relevant for PS version >= 7, the result can be irrelevant for older versions. See https://github.com/PowerShell/PowerShell/issues/12442.
+    if ($PSVersionTable.PSVersion.Major -lt 7) {
+        Say-Warning $diskSpaceWarning
```

Comment:
I wasn't sure about using `Warning` here because the behavior depends on the callers' warning preference, but the SDK Acquisition extension can control this on its end, so I think this is acceptable here :+1: